### PR TITLE
[RPA-1512] Tenants frontend implementation

### PR DESCRIPTION
**Pull request by Aeri4a**
## Description

image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

---

I agree to the terms of the RunBotics Contributor License Agreement.

### CODE REVIEW

File: runbotics/runbotics-common/src/model/api/tenant.model.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+export type UserDTO = Pick<IUser, 'id' | 'login'>;
+
+export interface Tenant {
+    id?: string;
+    name?: string;
+    createdBy?: UserDTO;
+    created?: string | null;
+    updated?: string | null;
+    lastModifiedBy?: string | null;
```
Comment (star-in-the-jar) [inline runbotics/runbotics-common/src/model/api/tenant.model.ts:9]: why can they be null and optional (?) parameter is not enough? Does it have anything to do with the DB?

File: runbotics/runbotics-common/src/model/api/tenant.model.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { IUser } from './user.model';
+
+export type UserDTO = Pick<IUser, 'id' | 'login'>;
```
Comment (star-in-the-jar) [inline runbotics/runbotics-common/src/model/api/tenant.model.ts]: Is userDTO sth we send to backend to add/modify/delete user?

File: runbotics/runbotics-orchestrator-ui/src/src-app/hooks/useTenantSearch.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            tenantsActions.getAllByPage({
+                page: newPage,
+                size: pageSize,
+                filter: {
+                    contains: { 'name': debouncedValue }
+                }
+            }))
+            .catch(() => {
+                enqueueSnackbar(
+                    translate('Tenants.List.View.Events.Error.FindingTenants'),
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/hooks/useTenantSearch.ts:61]: instead 'FindingTenants' you can use ${ERROR_KEY} from backend response If you have more reasons the error could occur. If that's the only one there's no need.

File: runbotics/runbotics-orchestrator-ui/src/src-app/translations/en/tenants/actions/modals.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+const modalsActionsTranslations = {
+    'Tenants.Actions.Modals.DeleteModal.TitleMessage': 'Are you sure you want to delete the following tenants?',
+    'Tenants.Actions.Modals.DeleteModal.Button.Delete': 'Confirm',
+    'Tenants.Actions.Modals.DeleteModal.Button.Cancel': 'Cancel',
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/translations/en/tenants/actions/modals.ts]: for those let's use translations from common.json

File: runbotics/runbotics-orchestrator-ui/src/src-app/translations/pl/tenants/actions/modals.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import englishModalsActionsTranslations from '#src-app/translations/en/tenants/actions/modals';
+
+const modalsActionsTranslations: typeof englishModalsActionsTranslations = {
+    'Tenants.Actions.Modals.DeleteModal.TitleMessage': 'Czy na pewno chcesz usunąć następujące organizacje?',
+    'Tenants.Actions.Modals.DeleteModal.Button.Delete': 'Potwierdź',
+    'Tenants.Actions.Modals.DeleteModal.Button.Cancel': 'Anuluj',
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/translations/pl/tenants/actions/modals.ts]: common.json again

File: runbotics/runbotics-orchestrator-ui/src/src-app/translations/pl/tenants/list/edit.ts [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import englishEditListTranslations from '#src-app/translations/en/tenants/list/edit';
+
+const editListTranslations: typeof englishEditListTranslations = {
+    'Tenants.List.Edit.Form.Title': 'Edytuj organizacje',
+    'Tenants.List.Edit.Form.Button.Delete': 'Usuń',
+    'Tenants.List.Edit.Form.Button.Cancel': 'Anuluj',
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/translations/pl/tenants/list/edit.ts]: common.json

File: runbotics/runbotics-orchestrator-ui/src/src-app/views/tenants/CreateTenantDialog.tsx [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+    open,
+    onClose
+}) => {
+    const { translate } = useTranslations();
+    const { enqueueSnackbar } = useSnackbar();
+    const dispatch = useDispatch();
+
+    const { refreshSearch } = useTenantSearch();
+    const [name, setName] = useState('');
+    const isFieldEmpty = !(name.trim() !== '');
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/views/tenants/CreateTenantDialog.tsx]: does it work for sure?

File: runbotics/runbotics-orchestrator-ui/src/src-app/views/tenants/DeleteTenantDialog.tsx [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+                );
+            });
+    };
+
+    useEffect(() => {
+        open && setTenantId(tenant.id);
+    }, [open]);
+
+    return (
+        <CustomDialog
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/views/tenants/DeleteTenantDialog.tsx:60]: cool, you've used that 😎

File: runbotics/runbotics-orchestrator-ui/src/src-app/views/tenants/TenantsListView/TenantsListView.tsx [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+        setIsEditDialogVisible(false);
+    };
+
+    const handleOpenCreateDialog = () => {
+        setIsCreateDialogVisible(true);
+    };
+
+    const handleCloseCreateDialog = () => {
+        setIsCreateDialogVisible(false);
+    };
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/views/tenants/TenantsListView/TenantsListView.tsx]: Close/Create dialog is quite well described. "handle" keyword is not neccessary here - it well describes general actions

File: runbotics-orchestrator/src/main/java/com/runbotics/service/impl/TenantServiceImpl.java [hunk 10]
```diff
@@ -102,20 +121,26 @@ public Optional<TenantDTO> partialUpdate(TenantDTO tenantDTO) {
...
-                                () -> new BadRequestAlertException("Could not found user while updating", ENTITY_NAME, "userNotFound")
-                            );
-                        tenant.setCreatedBy(updatedUser);
-                    }
-                    tenantDTO.setUpdated(ZonedDateTime.now());
-                    tenantMapper.partialUpdate(tenant, tenantDTO);
+            .map(tenant -> {
+                if (tenantDTO.getCreatedBy() != null) {
+                    final User updatedUser = userRepository.findById(tenantDTO.getCreatedBy().getId()).orElseThrow(
+                        () -> new BadRequestAlertException("Could not found user while updating", ENTITY_NAME, "userNotFound")
```
Comment (star-in-the-jar) [inline runbotics-orchestrator/src/main/java/com/runbotics/service/impl/TenantServiceImpl.java]: I haven't seen using "userNotFound" as translation key. Consider implementing this

**Review by star-in-the-jar — CHANGES_REQUESTED** [review]
bump version

File: runbotics/runbotics-common/src/model/api/tenant.model.ts (hunk 1 above)
Comment (Aeri4a) [reply runbotics/runbotics-common/src/model/api/tenant.model.ts:9]: This is in order to use this interface to update Tenant inside user - so we don't need all fields

File: runbotics/runbotics-orchestrator-ui/src/src-app/views/tenants/CreateTenantDialog.tsx (hunk 7 above)
Comment (Aeri4a) [reply runbotics/runbotics-orchestrator-ui/src/src-app/views/tenants/CreateTenantDialog.tsx]: Yup, but I will remove unnecessary '!' and reverse logic.

**Review by star-in-the-jar — APPROVED** [review]

**Pull request merged**